### PR TITLE
feature (ref 36294): find api user from list and unset that

### PR DESCRIPTION
**Pull request by jalalifardemos**
**Ticket:** https://yaits.demos-deutschland.de/T36294

Description: use condition to not show AiApiUser in the list

Delete the checkbox if it doesn't apply/isn't necessary.

- [ ] Tests updated/created
- [ ] Update documentation
- [ ] Link all relevant tickets
- [x] Move the tickets on the board accordingly

### CODE REVIEW

**Review by mussbach — CHANGES_REQUESTED** [review]

File: demosplan/DemosPlanCoreBundle/Controller/User/DemosPlanUserAPIController.php [hunk 1]
```diff
@@ -156,6 +157,14 @@ public function listAction(
                 $listResult = $jsonApiActionService->searchObjects($userType, $searchParams, $conditions, $sortMethods);
             }
             $users = $listResult->getList();
+            foreach ($users as $key => $user) {
```
[the remark this answers is not in the document]
Comment (jalalifardemos) [reply demosplan/DemosPlanCoreBundle/Controller/User/DemosPlanUserAPIController.php]: done in [ebfd9a1](https://github.com/demos-europe/demosplan-core/pull/2705/commits/ebfd9a19ded1c9422ab3934a6f843322ba6c65a8)

File: demosplan/DemosPlanCoreBundle/ResourceTypes/AdministratableUserResourceType.php [hunk 2]
```diff
@@ -77,6 +79,8 @@ protected function getAccessConditions(): array
             $this->conditionFactory->propertyHasValue(false, $this->deleted),
             // never show internal Citizen user
             $this->conditionFactory->propertyHasNotValue(User::ANONYMOUS_USER_ID, $this->id),
+            // never show ApiAiUser
+            $this->conditionFactory->propertyHasNotValue( AiApiUser::AI_API_USER_ID, $this->id),
```
Comment (mussbach) [inline demosplan/DemosPlanCoreBundle/ResourceTypes/AdministratableUserResourceType.php]: Could you please use AI_API_USER_LOGIN? This should be more stable

**Review by mussbach — CHANGES_REQUESTED** [review]

File: demosplan/DemosPlanCoreBundle/ResourceTypes/AdministratableUserResourceType.php (hunk 2 above)
Comment (jalalifardemos) [reply demosplan/DemosPlanCoreBundle/ResourceTypes/AdministratableUserResourceType.php]: done in [ad4f9a1](https://github.com/demos-europe/demosplan-core/pull/2705/commits/ad4f9a165ea864b05680491f786cbe5a862a83ed)

**Review by mussbach — APPROVED** [review]
lgm, thx, code review only

**Pull request merged**